index.bs

```diff
@@ -398,9 +398,15 @@ On getting, the {{NavigatorUAData/mobile}} attribute must return the [=user agen
 
 The <dfn method for="NavigatorUA"><code>getHighEntropyValues(|hints|)</code></dfn> method MUST run these steps:
 
-1. Let |p| be a [=a new promise=].
+1. Let |p| be a [=a new promise=] created with {{NavigatorUAData}}'s [=relevant realm=].
 
-2.  Run the following steps [=in parallel=]:
+2. If [=this=]'s [=relevant global object=] is not identical to {{NavigatorUAData}}'s [=relevant global object=], then:
```

Comment:
@dtapuska - Is this a good equivalent to `!GetExecutionContext()`?

Reply:
+ @domenic

I don't think this is quite right. Isn't this the same as NavigatorUAData?  I think what you mean is that the RelevantRealm != IncumbentRealm then throw WrongDocument ?

Reply:
Incumbent realm should not be used by new specs.

If you are trying to make sure that the document is alive, then you need to check if the relevant global object is a Window and its associated Document is fully active.

What are you actually trying to express here, in terms of developer-facing benefit, not in terms of Chromium implementation details?

Reply:
> If you are trying to make sure that the document is alive, then you need to check if the relevant global object is a Window and its associated Document is fully active.


Makes sense, thanks!

> What are you actually trying to express here, in terms of developer-facing benefit, not in terms of Chromium implementation details?

I'm trying to express that detached frames should reject here.
The developer-facing benefit is reduced confusion as the rest of UA data returns nullyfied values in that case.

Reply:
> the rest of UA data returns nullyfied values in that case.

That doesn't appear to be true, at least for anything in https://html.spec.whatwg.org/multipage/system-state.html#client-identification .

Reply:
In general I'd recommend against taking a dependency on global state in this way, and instead have these values be independent of such external factors.

Reply:
> That doesn't appear to be true, at least for anything in https://html.spec.whatwg.org/multipage/system-state.html#client-identification .

The plot thickens...
Chromium has a [test](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/fast/dom/Window/property-access-on-cached-properties-after-frame-removed.html?q=fast%2Fdom%2FWindow%2Fproperty-access-on-cached-properties-after-frame-removed.html&ss=chromium&originalUrl=https:%2F%2Fcs.chromium.org%2F) that enforces this.

I wrote a small [test page](https://yoav-scratchpad.glitch.me/ua_detached.html) for `navigator.userAgent` and it seems like Safari and Chromium both nullify the string, while Firefox does not...

Reply:
Ugh, that's bizarre. I hope we can align Chromium with the spec? There's just no reason to make these getters dependent on state like that.

Reply:
Maybe the easiest path forward would be to downsize this PR to the task queueing only, drop the Promise rejection, find a different ExecutionContext to queue the task on in the Chromium implementation (from the `ScriptState`) and open an issue on the HTML spec to figure out the nullification of attributes on detached documents.

Reply:
That makes sense to me.

Note that the spec as written rejects the promise but the rejection will never be seen, since the event loop is inactive for those documents.

Reply:
Removed the promise rejection, but left the realm on which the promise is created.

> Note that the spec as written rejects the promise but the rejection will never be seen, since the event loop is inactive for those documents.

Is this because of the realm the promise is created with?

Reply:
No, it's because the event loop just doesn't run for inactive documents.